src/raftstore/store/peer.rs

```diff
@@ -1668,12 +1668,20 @@ impl Peer {
         poll_ctx.raft_metrics.propose.read_index += 1;
 
         let renew_lease_time = monotonic_raw_now();
-        if let Some(read) = self.pending_reads.reads.back_mut() {
-            if read.renew_lease_time + poll_ctx.cfg.raft_store_max_leader_lease() > renew_lease_time
-            {
-                read.cmds.push((req, cb));
-                return false;
+        match self.inspect_lease() {
+            LeaseState::Valid | LeaseState::Expired => {
```

Comment:
May need to write comments explaining why it's OK to check the current state instead of including state when the previous heartbeat is sent.